lib/kafka/connection.rb

```diff
@@ -75,25 +75,27 @@ def close
     #
     # @return [Object] the response.
     def send_request(request)
-      # Default notification payload.
-      notification = {
-        broker_host: @host,
-        api: Protocol.api_name(request.api_key),
-        request_size: 0,
-        response_size: 0,
-      }
+      with_retry do
+        # Default notification payload.
+        notification = {
+          broker_host: @host,
+          api: Protocol.api_name(request.api_key),
+          request_size: 0,
+          response_size: 0,
+        }
 
-      @instrumenter.instrument("request.connection", notification) do
-        open unless open?
+        @instrumenter.instrument("request.connection", notification) do
+          open unless open?
 
-        @correlation_id += 1
+          @correlation_id += 1
 
-        write_request(request, notification)
+          write_request(request, notification)
 
-        response_class = request.response_class
-        wait_for_response(response_class, notification) unless response_class.nil?
+          response_class = request.response_class
+          wait_for_response(response_class, notification) unless response_class.nil?
+        end
       end
-    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT, EOFError => e
+    rescue Errno::ETIMEDOUT => e
```

Comment:
I have a small question about this, why don't rescue all the same errors as before? 

We were seeing in of our service a high amount of connection errors:
```
 Could not connect to broker kafka-3 (node_id=301): Connection error: EOFError
``` 

And that was having a big impact on the service throughput.  I saw your PR and we tried adding this retry and the connection errors dropped completely  ❤️  .  But I kept the old rescue, since i'm not sure about the impact of changing the rescue

```
rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT, EOFError => e
```